File: CodeChallanges3/app/src/main/java/com/example/codechallanges3/Adapters/RecyclerViewAdapter.java
package com.example.codechallanges3.Adapters;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.codechallanges3.R;

import java.util.List;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.RecycleViewHolder> {
    private final List<String> firstNames;
    private final List<String> lastNames;

    public RecyclerViewAdapter(List<String> firstData,List<String> secondData) {
        this.firstNames = firstData;
        this.lastNames = secondData;
    }

    @NonNull
    @Override
    public RecyclerViewAdapter.RecycleViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View recyclerRow = layoutInflater.inflate(R.layout.item_student_name, parent, false);
        return new RecycleViewHolder(recyclerRow);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerViewAdapter.RecycleViewHolder holder, int position) {
        String firstName=firstNames.get(position);
        String lastName=lastNames.get(position);
        holder.studentFirstName.setText(firstName);
        holder.studentLastName.setText(lastName);
        if(position%2==1){
            holder.layout.setBackgroundColor(Color.parseColor("#D3D3D3"));
        }else{
            holder.layout.setBackgroundColor(Color.parseColor("#819ca9"));
        }
    }

    @Override
    public int getItemCount() {
        return firstNames.size();
    }

    static class RecycleViewHolder extends RecyclerView.ViewHolder {
        final TextView studentFirstName;
        final TextView studentLastName;
        final LinearLayout layout;

        RecycleViewHolder(@NonNull View itemView) {
            super(itemView);
            studentFirstName = itemView.findViewById(R.id.student_first_name);
            studentLastName = itemView.findViewById(R.id.student_last_name);
            layout = itemView.findViewById(R.id.list_item);
        }
    }
}

File: GameCollector/app/src/main/java/com/example/gamecollector/adapters/RecyclerViewAdapter.kt
package com.example.gamecollector.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AnimationUtils
import android.widget.AdapterView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.gamecollector.R
import com.example.gamecollector.activities.GamePageActivity
import com.example.gamecollector.data.Game
import kotlinx.android.synthetic.main.item_game.view.*

class RecyclerViewAdapter(var context: Context,val itemClickListener: OnItemClickListener) :
    RecyclerView.Adapter<RecyclerViewAdapter.RecycleViewHolder>() {

    var gameList = emptyList<Game>()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): RecycleViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_game, parent, false)
        return RecycleViewHolder(view)
    }

    override fun getItemCount() = gameList.size

    override fun onBindViewHolder(holder: RecycleViewHolder, position: Int) {
        holder.container.animation =
            AnimationUtils.loadAnimation(context, R.anim.game_card_animation)
        val item = gameList[position]
        holder.gameTitle.text = item.title
        holder.gameRating.text = item.rating.toString()
        holder.bind(gameList[position],itemClickListener)
    }

    inner class RecycleViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val gameTitle: TextView = view.item_game_title
        val gameRating: TextView = view.item_game_rating
        val gameGenre: TextView = view.item_game_platform
        val container: LinearLayout = view.game_item

        fun bind(game: Game,clickListener: OnItemClickListener)
        {
            gameTitle.text = game.title
            gameRating.text = game.rating.toString()
            gameGenre.text = game.genre

            itemView.setOnClickListener {
                clickListener.onItemClicked(game)

            }
        }
    }

    private var onItemClick: View.OnClickListener

    init {
        onItemClick = View.OnClickListener {
            val item = it.tag as? Game
            val intent = Intent(context, GamePageActivity::class.java)
            intent.putExtra("Title", item?.title)
            intent.putExtra("Description", item?.description)
            intent.putExtra("Rating", item?.rating)
            intent.putExtra("Genre", item?.genre)
            intent.putExtra("Year", item?.year)
            it.context.startActivity(intent)
        }

    }
}


interface OnItemClickListener{
    fun onItemClicked(game: Game)
}